src/sentry/plugins/sentry_mail/activity/release.py

```diff
@@ -100,20 +100,38 @@ def get_participants(self):
         if not self.email_list:
             return {}
 
-        # identify members which have been seen in the commit log and have
-        # verified the matching email address
-        return {
-            user: GroupSubscriptionReason.committed
+        # collect all users with verified emails on a team in the related projects,
+        # and couple them with the the users' setting value for deploy-emails
+        users_with_options = [
```

Comment:
good point, this code path could multiply out to be a ton of queries right now so I'll swap it out for your approach.